### ISSUE TITLE
TuYa TS0601 Air quality Sensor : unit error ?

### ISSUE THREAD

**Issue by tazelbaze** [Koenkk/zigbee2mqtt#21666]
### What happened?

in the descriptive note, the units given for this sensor are:
- VOC: ppm
- formaldehyd: mg/m3

However, in the Tuya application, these two data appear:
- VOC: with a factor /100 (0.10 ppm instead of 10 ppm)
- formaldehyd: µg/m³ and not g/m3

Given the role of these sensors, these differences are decisive because if the units are correct then the air is very dangerous (including within the given measurement range).

https://www.zigbee2mqtt.io/devices/TS0601_air_quality_sensor.html#tuya-ts0601_air_quality_sensor

Ce sujet aborde le problème mais très partiellement et sans correction réelle :
https://github.com/zigpy/zha-device-handlers/issues/953

### What did you expect to happen?

Identify and check if the units are correct

### How to reproduce it (minimal and precise)

i have two devices. Same conclusions.

### Zigbee2MQTT version

1.36.0-1

### Adapter firmware version

ca5298d4c512ec11b93121c7bd930c07

### Adapter

usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus

### Setup

Add-on Home-Assistant on RPI4

### Debug log

[45df7312_zigbee2mqtt_2024-03-03T12-59-27.425Z.log](https://github.com/Koenkk/zigbee2mqtt/files/14473021/45df7312_zigbee2mqtt_2024-03-03T12-59-27.425Z.log)

**Comment by maxevilmind** [comment]
Yeah this sensor data does not make any sense, if I look at the numbers I should have been dead already. Especially formaldehyde levels

**Comment by ArGGu** [comment]
I got "_TZE204_yvx5lh6k" and definitely the VOC and formaldehyd values are not correct. I have not used with Tuya, so I can't compare to those. But I bought one cheap air quality sensor that has display on it and comparing values if the VOC is divided with 100 and formaldehyd is µg/m³ then the value are much closer. Though the cheap one with the display is still showing way lower values.

But at least dividing the VOC with 100 and using µg/m³ for formaldehyd then values would make sense. Both sensors I got are cheap, so probably not that accurate. But also this device has a lot of fingerprints defined in the converter, so does all of them report the values in the same units.

**Comment by mayraost** [comment]
Still an issue https://github.com/Koenkk/zigbee2mqtt/issues/27205

**Comment by krzkraw** [comment]
This should be fixed now with [commit in converters](https://github.com/Koenkk/zigbee-herdsman-converters/commit/e31057010670dd6a8cf14b78e1b8a11ea826b9c0)
